query-engine/core/src/interpreter/query_interpreters/inmemory_record_processor.rs

```diff
@@ -193,6 +193,6 @@ impl InMemoryRecordProcessor {
     }
 
     fn must_apply_pagination(&self) -> bool {
-        self.take.or(self.skip).is_some() || self.cursor.is_some()
+        self.take.is_some() || self.skip.is_some() || self.cursor.is_some()
```

Comment:
This technically changes the semantics a tiny bit, but I think it is safe and can't lead to a difference in behaviour in practice. Whether we apply in-memory pagination here only matters if we also use in-memory distinct, and that, in turn, only matters if we have more than one record.

Reply:
Yes, we have the `Some` and `One` cases separated and pagination does not work for `One` anymore. But it should not be an issue in practice, as you pointed out. 

This confusion is caused by pagination not being represented clearly in the data model, which should be fixed by later refactoring.